thorlcr/activities/firstn/thfirstnslave.cpp

```diff
@@ -294,14 +298,7 @@ class CFirstNSlaveGlobal : public CFirstNSlaveBase, implements ILookAheadStopNot
             if (limit+skipCount<r)
                 r = limit+skipCount;
             // sneaky short circuit
-            {
-                CriticalBlock b(crit);
-                if (RCUNBOUND == maxres)
-                {
-                    maxres = r;
-                    sendCount();
-                }
-            }
+            sendOnce(r);
```

Comment:
It would only send if maxres was not set before, i.e. if RCUNBOUND == maxres).
The same is true now, sendOnce is called and in sendOnce and in there, if maxres != RCUNBOUND it returns before sending.

Reply:
You're right I misread the code